pkg/agent/params.go

```diff
@@ -94,6 +93,34 @@ func getParameters(config *datamodel.NodeBootstrappingConfiguration, generatorCo
 			addValue(parametersMap, fmt.Sprintf("%sosImageSKU", agentProfile.Name), cloudSpecConfig.OSImageConfig[datamodel.Distro(agentProfile.Distro)].ImageSku)
 			addValue(parametersMap, fmt.Sprintf("%sosImagePublisher", agentProfile.Name), cloudSpecConfig.OSImageConfig[datamodel.Distro(agentProfile.Distro)].ImagePublisher)
 			addValue(parametersMap, fmt.Sprintf("%sosImageVersion", agentProfile.Name), cloudSpecConfig.OSImageConfig[datamodel.Distro(agentProfile.Distro)].ImageVersion)
+		} else {
+			// Set ImageRef if it is not nil and always set the Windows VHD information in WindowsProfile.
+			// ImageRef will be used to generate ARM template for the agent pool if it is set.
+			// Otherwise, the Windows VHD information in WindowsProfile will be used to generate ARM template.
+			// Priority:
+			//   1. ImageRef in agent pool
+			//   2. ImageRef in WindowsProfile
+			//   3. PIR image in WindowsProfile
+			if agentProfile.ImageRef != nil {
+				addValue(parametersMap, fmt.Sprintf("%sosImageName", agentProfile.Name), agentProfile.ImageRef.Name)
+				addValue(parametersMap, fmt.Sprintf("%sosImageResourceGroup", agentProfile.Name), agentProfile.ImageRef.ResourceGroup)
+			} else if properties.WindowsProfile.HasImageRef() {
+				addValue(parametersMap, fmt.Sprintf("%sosImageName", agentProfile.Name), properties.WindowsProfile.ImageRef.Name)
+				addValue(parametersMap, fmt.Sprintf("%sosImageResourceGroup", agentProfile.Name), properties.WindowsProfile.ImageRef.ResourceGroup)
```

Comment:
I don't like the fact that we still read imageRef from windows profile - it should be passed in in the agent pool profile from RP side. Can we fix that?